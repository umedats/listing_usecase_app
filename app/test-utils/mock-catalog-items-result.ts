export const MOCK_CATALOG_ITEMS_API_RESPONSE: object = {
  numberOfResults: 1,
  items: [
    {
      asin: "B00004SPEC",
      productTypes: [
        {
          marketplaceId: "ATVPDKIKX0DER",
          productType: "KITCHEN",
        },
      ],
      summaries: [
        {
          marketplaceId: "ATVPDKIKX0DER",
          adultProduct: false,
          autographed: false,
          brand: "KRUPS",
          browseClassification: {
            displayName: "Countertop Blenders",
            classificationId: "289915",
          },
          itemClassification: "BASE_PRODUCT",
          itemName: "Krups 243-70 Power X Plus Combi Blender, DISCONTINUED",
          manufacturer: "KRUPS",
          memorabilia: false,
          modelNumber: "243-70",
          packageQuantity: 1,
          partNumber: "243-70",
          style: "Modern",
          tradeInEligible: false,
          websiteDisplayGroup: "kitchen_display_on_website",
          websiteDisplayGroupName: "Kitchen",
        },
      ],
    },
  ],
};

export const MOCK_CATALOG_ITEMS_API_EMPTY_RESPONSE: object = {
  numberOfResults: 0,
};
