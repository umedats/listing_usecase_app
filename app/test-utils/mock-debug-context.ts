import { SPAPIRequestResponse } from "@/app/model/types";

export const MOCK_SP_API_RESPONSE: SPAPIRequestResponse = {
  apiName: "GetListingsItem",
  apiDocumentationLink: "https://amazon.com",
  request: {
    marketplaceId: "ATVPDKIKX0DER",
    locale: "en_US",
    asin: "A1F83G8C2ARO7P",
  },
  response: {
    item: "Dummy Item",
  },
};
