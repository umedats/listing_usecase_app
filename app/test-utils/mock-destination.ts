import {
  CreateDestinationResponse,
  Destination,
} from "@/app/sdk/notifications";

export const MOCK_DESTINATION: Destination = {
  name: "TestDestination",
  destinationId: "5fe08e61-c5ec-4427-9f25-24a7b630cce2",
  resource: {
    eventBridge: {
      name: "sellingpartnerapi.amazon.com/amzn1.sellerapps.app.15a75829-cd4a-4efc-b947-0cc39d874577",
      region: "us-east-1",
      accountId: "123456",
    },
  },
};

export const MOCK_CREATE_DESTINATION_RESPONSE: CreateDestinationResponse = {
  payload: MOCK_DESTINATION,
  errors: {},
};
