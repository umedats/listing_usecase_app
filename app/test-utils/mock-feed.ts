export const MOCK_FEED_1 = {
  feedId: "1000",
  feedType: "JSON_LISTINGS_FEED",
  createdTime: "2023-10-05T04:23:58+0000",
  processingStatus: "IN_PROGRESS",
  resultFeedDocumentId:
    "amzn1.tortuga.3.ed4cd0d8-447b-4c22-96b5-52da8ace1207.T3YUVYPGKE9BMY",
};

export const MOCK_FEED_2 = {
  feedId: "2000",
  feedType: "JSON_LISTINGS_FEED",
  createdTime: "2023-10-05T04:23:58+0000",
  processingStatus: "IN_PROGRESS",
};

export const MOCK_CREATE_FEED_DOCUMENT_SUCCESS = {
  feedDocumentId: "123456",
  url: "https:feed.document/url",
};

export const MOCK_CREATE_FEED_SUCCESS = {
  feedId: "123456789",
};
