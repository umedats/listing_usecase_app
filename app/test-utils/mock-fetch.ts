import fetch from "jest-fetch-mock";

/**
 * Mocks a fetch call with the Promise.resolve().
 * @param status the HTTP status of the mocked response.
 * @param body any body of the mocked response.
 */
export function mockResolveFetchResponse(status: number, body: object) {
  fetch.mockResponse(() =>
    Promise.resolve({
      status: status,
      body: JSON.stringify(body),
    }),
  );
}

/**
 * Mocks a fetch call with the Promise.resolve(). The given body is not
 * serialized into Json.
 * @param status the HTTP status of the mocked response.
 * @param body any body of the mocked response.
 * @param headers the headers for the response.
 */
export function mockResolveFetchResponseNoJsonSerialization(
  status: number,
  body: any,
  headers?: any,
) {
  fetch.mockResponse(() =>
    Promise.resolve({
      status: status,
      body: body,
      headers: headers,
    }),
  );
}

/**
 * Mocks a fetch call with the Promise.reject().
 * @param status the HTTP status of the mocked response.
 */
export function mockRejectFetchResponse(status: number) {
  fetch.mockResponse(() =>
    Promise.reject({
      status: status,
    }),
  );
}

/**
 * Mocks a fetch response by introducing a delay.
 * @param status the HTTP response status
 * @param body the body of the response
 * @param delayInMillis the delay for the promise resolution.
 */
export function mockDelayedResolveFetchResponse(
  status: number,
  body: object,
  delayInMillis: number,
) {
  const delayedPromise = new Promise((resolve) => {
    setTimeout(
      () =>
        resolve({
          status: status,
          body: JSON.stringify(body),
        }),
      delayInMillis,
    );
  });
  // @ts-ignore
  fetch.mockResponse(() => delayedPromise);
}
