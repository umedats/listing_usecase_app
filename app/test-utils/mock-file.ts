import { JSON_FILE_TYPE } from "@/app/constants/global";

const JSON_FEED_NAME = "feed.json";

export function createFile(
  fileContent: string,
  fileName: string,
  fileType: string,
) {
  const file = new File([fileContent], fileName, { type: fileType });
  file.text = () =>
    new Promise((resolve) => {
      resolve(fileContent);
    });
  return file;
}

export function createEmptyJsonFile(): File {
  return createFile("", JSON_FEED_NAME, JSON_FILE_TYPE);
}

export function createJsonFile(fileContent: string): File {
  return createFile(fileContent, JSON_FEED_NAME, JSON_FILE_TYPE);
}
