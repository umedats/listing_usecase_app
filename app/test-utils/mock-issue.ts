import { Issue } from "@/app/model/types";

export const MOCK_ISSUE: Issue = {
  code: "9001",
  message: "The value is invalid.",
  severity: "ERROR",
};

export const MOCK_ISSUE_WITH_ATTR_NAMES: Issue = {
  code: "9001",
  message: "The value is invalid.",
  severity: "ERROR",
  attributeNames: ["item_name", "description"],
};
