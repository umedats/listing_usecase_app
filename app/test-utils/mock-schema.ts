/**
 * Mock Schema which can be used across tests. The schema will be filled
 * with more content as needed.
 */
export const MOCK_SCHEMA = {
  $schema:
    "https://schemas.amazon.com/selling-partners/definitions/product-types/meta-schema/v1",
  $id: "https://schemas.amazon.com/selling-partners/definitions/product-types/schema/v1/LUGGAGE",
  properties: {
    name: {
      type: "string",
      minLength: 3,
      description: "Please enter your name",
    },
    birthDate: {
      type: "string",
      format: "date",
    },
    occupation: {
      type: "string",
    },
  },
  required: ["name", "birthDate"],
  anyOf: [
    {
      type: "string",
    },
    {
      type: "string",
      enum: ["ATVPDKIKX0DER"],
      enumNames: ["Amazon.com"],
    },
  ],
  allOf: [
    {
      required: ["fulfillment_channel_code"],
      properties: {
        fulfillment_channel_code: {
          enum: ["DEFAULT"],
        },
      },
    },
    {
      not: {
        required: ["quantity"],
      },
    },
  ],
  oneOf: [
    {
      type: "string",
    },
    {
      type: "string",
      enum: ["ATVPDKIKX0DER"],
      enumNames: ["Amazon.com"],
    },
  ],
};
