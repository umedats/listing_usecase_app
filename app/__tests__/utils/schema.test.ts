import {
  cleanUpListing,
  cleanUpSchemaForPresentation,
} from "@/app/utils/schema";
import { MOCK_SCHEMA } from "@/app/test-utils/mock-schema";

describe("Test for the cleanUpSchemaForPresentation", () => {
  test("verifies if all the combinator keys are removed.", () => {
    const cleanedUpSchema = cleanUpSchemaForPresentation(MOCK_SCHEMA);
    expect(cleanedUpSchema.anyOf).toBeUndefined();
    expect(cleanedUpSchema.allOf).toBeUndefined();
    expect(cleanedUpSchema.oneOf).toBeUndefined();
    expect(cleanedUpSchema.$id).toContain("Augmented");
  });
});

describe("Test for the cleanUpListing", () => {
  test("snapshot test for the cleanUpListing", () => {
    const listing = {
      item_name: [
        {
          value: "",
          marketplace_id: undefined,
        },
      ],
      description: [
        {
          value: {
            marketplace_id: undefined,
          },
          alternateValue: {
            value: 1234,
          },
        },
      ],
    };
    expect(cleanUpListing(listing)).toMatchSnapshot();
  });
});
