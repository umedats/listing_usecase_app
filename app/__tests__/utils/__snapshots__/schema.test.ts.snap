// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the cleanUpListing snapshot test for the cleanUpListing 1`] = `
{
  "description": [
    {
      "alternateValue": {
        "value": 1234,
      },
    },
  ],
}
`;
